### PR TITLE
Save credentials as secrets for globaldnsprovider

### PR DESCRIPTION
**Pull request by prachidamle**
https://github.com/rancher/rancher/issues/21369

### CODE REVIEW

**Comment by alena1108** [comment]
@prachidamle have we run the backwards compatibility test? Also can we add an integration/unit test for it. I believe there is a way to make direct api calls to bypass rancher apis in integration tests (@mrajashree did it for one of her prs); this way you can fetch global dns crd and check that the field has become a secret ref.

**Comment by prachidamle** [comment]
@alena1108 yes ran the backwards compatibility test. On existing crds, any update to the globaldnsprovider will turn the keys to secrets and deploy new pod.

**Comment by kinarashah** [comment]
> @prachidamle have we run the backwards compatibility test? Also can we add an integration/unit test for it. I believe there is a way to make direct api calls to bypass rancher apis in integration tests (@mrajashree did it for one of her prs); this way you can fetch global dns crd and check that the field has become a secret ref.

Just for reference, `integration/suite/test_password_store.py` already has similar tests for logging configs

**Comment by alena1108** [comment]
LGTM, merge after integration test is in

File: pkg/controllers/management/globaldns/globaldnsprovider_secret_syncer.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+//sync is called periodically and on real updates
+func (n *ProviderSecretSyncer) sync(key string, obj *corev1.Secret) (runtime.Object, error) {
+	if obj == nil || obj.DeletionTimestamp != nil {
+		return nil, nil
+	}
+	return n.reconcileAllGlobalDNSProviders(key)
+}
+
+func (n *ProviderSecretSyncer) reconcileAllGlobalDNSProviders(key string) (runtime.Object, error) {
+	globalDNSProviderList, err := n.globalDNSProviderLister.List(namespace.GlobalNamespace, labels.NewSelector())
```
Comment (alena1108) [inline pkg/controllers/management/globaldns/globaldnsprovider_secret_syncer.go]: @prachidamle you should only reconcile if the secret belongs to the cattle global namespace

File: pkg/api/store/globaldns/globaldnsprovider_store.go [hunk 2]
```diff
@@ -46,3 +52,59 @@ func canDeleteProvider(apiContext *types.APIContext, id string) error {
...
+
+	if !passwordFound {
+		existingProvider, err := p.ByID(apiContext, schema, id)
+		if err != nil {
+			return nil, err
+		}
+		existingPassword, _ := values.GetValue(existingProvider, keys...)
+		if !strings.HasPrefix(convert.ToString(existingPassword), namespace.GlobalNamespace) {
+			values.PutValue(data, existingPassword, keys...)
+			logrus.Infof("ProviderPwdStore updated data %v", data)
```
Comment (alena1108) [inline pkg/api/store/globaldns/globaldnsprovider_store.go]: better log it in Debug

Comment (prachidamle) [reply pkg/api/store/globaldns/globaldnsprovider_store.go]: removed it, had added for my debugging

**Comment by alena1108** [comment]
LGTM

**Pull request merged**